### PR TITLE
work on progress.. task1 js3

### PR DESCRIPTION
**Pull request by claudion3**

### CODE REVIEW

File: javascript/javascript3/week1/Giphy_api/giphy.html [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+<body>
+    <div class="app-wrap">
+        <header>
+            <h1>Giphy App</h1>
+        </header>
+        <main>
+            <input type="text" class="search-gifs" placeholder="search for gifs...">
+            <input type="text" class="number" placeholder="number">
+            <div class="check-gifs" id="start">
+                <button>Search for gifs</button>
```
Comment (piyushpalod) [inline javascript/javascript3/week1/Giphy_api/giphy.html:20]: button type should be submit

File: javascript/javascript3/week1/Giphy_api/giphy.html [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    </div>
+
+
+
+
+
+
+
+    <script>
+        const button = document.querySelector("button");
```
Comment (piyushpalod) [inline javascript/javascript3/week1/Giphy_api/giphy.html:38]: You can separate out your business logic to JS file

Comment (piyushpalod) [inline javascript/javascript3/week1/Giphy_api/giphy.html:38]: Variable name can be improved. btnSubmit ?

File: javascript/javascript3/week1/Giphy_api/giphy.html [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        let inputGifs = document.querySelector(".search-gifs");
+        let inputNumber = document.querySelector(".number");
+        let getInputGif, getInputNumber;
+        inputGifs.oninput = getGifs;
+        function getGifs(gif) {
```
Comment (piyushpalod) [inline javascript/javascript3/week1/Giphy_api/giphy.html:43]: No need to make separate function if calling only one place, use direct value in URL

File: javascript/javascript3/week1/Giphy_api/giphy.html [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        }
+
+
+
+        function displayResults(DataGif) {
+
+            let checkInputNumb = getInputNumber
+            console.log(checkInputNumb)
+            if (checkInputNumb === undefined) {
+                for (let i = 0; i < limite; i++) {
```
Comment (piyushpalod) [inline javascript/javascript3/week1/Giphy_api/giphy.html:77]: You can use forEach() instant of for()

File: javascript/javascript3/week1/Giphy_api/giphy.html [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                .then(displayResults);
+        }
+
+
+
+        function displayResults(DataGif) {
+
+            let checkInputNumb = getInputNumber
+            console.log(checkInputNumb)
+            if (checkInputNumb === undefined) {
```
Comment (piyushpalod) [inline javascript/javascript3/week1/Giphy_api/giphy.html:76]: I guess you want to check here if not null or undefined ?

File: javascript/javascript3/week1/Json_file/cph_restaurants.json [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+				"street name": "Esplanaden",
+				"street number": 8,
+				"zip": 1263,
+				"city": "København"
+			}
+		],
+		"tel": 33939844,
+		"website": "http://bistroboheme.dk"
+	}
+]
```
Comment (piyushpalod) [inline javascript/javascript3/week1/Json_file/cph_restaurants.json:159]: Use JSON beautifier/validator to format it.

File: javascript/javascript3/week1/weather_app/main.js [hunk 7]
```diff
@@ -0,0 +1,195 @@
+(function () {
```
Comment (piyushpalod) [inline javascript/javascript3/week1/weather_app/main.js:1]: Can you please explain this, whats happening ? You can make it more simple

File: javascript/javascript3/week1/weather_app/main.js [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+	return r;
+})()(
+	{
+		1: [
+			function (require, module, exports) {
+				(function (global) {
+					"use strict";
+
+					// ref: https://github.com/tc39/proposal-global
+					var getGlobal = function () {
```
Comment (piyushpalod) [inline javascript/javascript3/week1/weather_app/main.js:46]: why you need to check all this conditions ?

File: javascript/javascript3/week1/Giphy_api/giphy.html (hunk 3 above)
Comment (claudion3) [reply javascript/javascript3/week1/Giphy_api/giphy.html:43]: i'm calling two place (gifs type and number )

**Pull request merged**